docs/src/snap/howto/restore-quorum.md

```diff
@@ -0,0 +1,154 @@
+
+Be aware that Canonical Kubernetes uses not one, but two dqlite databases:
+
+* k8s-dqlite - used by Kubernetes itself
+* k8sd - Kuberenetes cluster management data
+
+Each database has its own state directory:
+
+* ``/var/snap/k8s/common/var/lib/k8s-dqlite``
+* ``/var/snap/k8s/common/var/lib/k8sd/state``
+
+The state directory normally contains:
+
+* ``info.yaml`` - the id, address and cluster role of this node
+* ``cluster.yaml`` - the state of the cluster, as seen by this dqlite node.
+  It includes the same information as info.yaml, but for all cluster nodes.
+* ``00000abcxx-00000abcxx``, ``open-abc`` - database segments
+* ``cluster.crt``, ``cluster.key`` - node certificates
+* ``snapshot-abc-abc-abc.meta``
+* ``metadata{1,2}``
+* ``*.sock`` - control unix sockets
+
+K8sd, which leverages the microcluster library, additionally contains:
+
+* ``server.crt``, ``server.key`` certificates
+* ``truststore`` folder, containing trusted certificates
+* ``daemon.yaml`` - k8sd daemon configuration
+* separate ``database`` folder
+
+Dqlite cluster members have one of the following roles:
+
+* 0 (voter) - Voting member of the database. A majority of voters is a quorum.
+* 1 (stand-by) - Non-voting member of the database; can be promoted to voter.
+* 2 (spare) - Not a member of the database.
+
+## Stop Canonical K8s services on all nodes
+
+Before attempting to recover the cluster, all remaining Canonical K8s services
+must be stopped. Use the following command on every node:
+
+```
+sudo snap stop k8s
+```
+
+## Reconfiguring the database
+
+Out of the remaining cluster nodes that are still alive, we must pick the
+one that has the most recent version of the Raft log.
+
+We need to update the ``cluster.yaml`` files, changing the role of the lost
+nodes to "spare" (2). Additionally, double check the addresses and IDs specified
+in ``cluster.yaml``, ``info.yaml`` and ``daemon.yaml``, especially if database
+files were moved across nodes.
+
+The following command can guide us through the process, prompting a text editor
+with informative inline comments for each of the dqlite configuration file.
+
+```
+sudo /snap/k8s/current/bin/k8sd cluster-recover \
+    --state-dir=/var/snap/k8s/common/var/lib/k8sd/state \
+    --k8s-dqlite-state-dir=/var/snap/k8s/common/var/lib/k8s-dqlite \
+    --log-level 0
+```
+
+Feel free to increase the log level for additional debug messages. Also note
+that the command creates database backups before making any changes.
+
+Once the Dqlite configuration is updated, the above command will reconfigure
+the Raft members.
+
+Finally, the command creates recovery tarballs that can be used to restore the
+lost nodes.
+
+Once the "cluster-recover" command completes, go ahead and restart the k8s
+services on this node:
+
+```
+sudo snap start k8s
+```
+
+Make sure that the services started successfully by using ``sudo snap services k8s``.
+In a few moments, the ``k8s status`` command will report the cluster as ready.
```

Comment:
nit: you can use `--wait-ready` to actually let the command wait until the node is ready.